src/lib.rs

```diff
@@ -41,16 +41,21 @@ pub fn from_hex(v: String) -> Result<Vec<u8>, Error> {
 /// Derives a public DH key from a static DH secret.
 /// sk must be 64 bytes in length or an error will be returned.
 pub fn public_key_from_secret(sk: Vec<u8>) -> Result<Vec<u8>, Error> {
-    if sk.len() != 64 {
-        return Err(Error::new("Secret key must be 64 bytes"));
-    }
-    let sec_key =
-        SecretKey::from_ed25519_bytes(sk.as_slice()).map_err(|err| Error::new(&err.to_string()))?;
-    let pair = sr25519::Pair::from(sec_key);
+    let pair = sr25519_keypair_from_secret_key(sk)?;
     let ss = derive_static_secret(&pair);
     Ok(PublicKey::from(&ss).as_bytes().to_vec())
 }
 
+fn sr25519_keypair_from_secret_key(secret_key_vec: Vec<u8>) -> Result<sr25519::Pair, Error> {
+    if secret_key_vec.len() != 64 {
+        return Err(Error::new("Secret key must be 64 bytes"));
```

Comment:
As a sidenote, we could probably add a `FromStr` impl for the error type to clean this up a tiny bit